migrate paloaltonetworks to typespec

welovej opened the pull request:
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

![spec_pr_review_workflow_diagram](https://github.com/Azure/azure-rest-api-specs/assets/4429827/5bb5e7ce-8aff-4dbb-a3f8-0d9b68fef5b1)

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [ ] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [x] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [ ] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [ ] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.
- [ ] A [release plan](https://aka.ms/azsdkdocs/release-plans) has been created. If not, please create one as it will help guide you through the REST API and SDK creation process. 

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.
- For guidance on SDK breaking change review, refer to https://aka.ms/ci-fix.

welovej committed:
converted

welovej committed:
compile successfully

welovej committed:
update

welovej committed:
fix for swagger validation:　GlobalRulestack_List_MaximumSet_Gen

welovej committed:
update

welovej committed:
update

welovej committed:
update

welovej committed:
fix for 404

welovej committed:
prettier

welovej committed:
supress warning

welovej committed:
update

welovej committed:
Update tspconfig.yaml

welovej committed:
Update tspconfig.yaml

welovej committed:
Merge remote-tracking branch 'upstream/main' into TspMig-paloaltonetworks

welovej committed:
justification and recovery of opeartionid

welovej committed:
Update models.tsp

welovej committed:
Merge branch 'main' into TspMig-paloaltonetworks

welovej committed:
update

welovej committed:
update

welovej committed:
update

welovej committed:
Merge branch 'main' into TspMig-paloaltonetworks

Prakriti-nith reviewed (commented):
The tag VersioningReviewRequired indicates that changes include breaking updates. In such cases, you should either introduce a new version or remove all breaking changes. If all breaking changes are removed, this tag will likely not be present.

Additionally, there are a lot of PR checks failing-
1. Breaking changes failures - Ideally typespec generation should not have any breaking changes. Why are these introduced? Can we please address and fix these?
2. Lint failures - Can you please address all the lint failures too?
3. SDK generation failures for go/java/others - Can you please address all these failures too? 
4. typespec validation check is also failing. Please address this too

Please address these failures and review questions before I review the full PR and test these on canary

Also, will the current PR be merged into the main branch, or do we need to create a new PR for that? The 2nd and 3rd points in the ICM are a bit unclear. 

welovej committed:
Update tspconfig.yaml

welovej committed:
Update tspconfig.yaml

welovej committed:
fix

tadelesh committed:
fix go config

XiaofeiCao committed:
unblock java automation

weidongxu-microsoft committed:
Apply TypeSpec migration compatibility fixes

- Fixed model name compatibility: IPAddress->IpAddress, IPAddressSpace->IpAddressSpace, DNSSettings->DnsSettings, DNSProxy->DnsProxy, EnabledDNSType->EnabledDnsType
- Updated service name configuration for manager consistency
- Removed invalid version dependencies from main.tsp

msyyc committed:
Update tspconfig.yaml

tadelesh committed:
change go ns

msyyc committed:
Merge branch 'main' into TspMig-paloaltonetworks

msyyc committed:
fix rename breaking for python

v-jiaodi committed:
Update PaloAltoNetworks configuration for SDK

v-jiaodi committed:
Remove emitter-output-dir from tspconfig.yaml

Removed emitter-output-dir from configuration.

v-jiaodi committed:
Add typespec-title-map for CloudngfwClient

v-jiaodi committed:
Modify emitter output directory in tspconfig.yaml

Update emitter output directory for Azure tools.

welovej committed:
Merge remote-tracking branch 'upstream/main' into TspMig-paloaltonetworks

welovej committed:
swagger recovery of version v2025_07_07_preview

welovej committed:
update version to 2025-10-08

welovej committed:
Update cspell.yaml

welovej committed:
Update cspell.yaml

welovej committed:
Revert "Update cspell.yaml"

This reverts commit 55728175093abc29efb83a45092adcd9c7f3a728.

welovej committed:
Update cspell.yaml

welovej committed:
update: ArmDeletedNoContentResponse --> ArmNoContentResponse

pshao25 committed:
Merge branch 'main' into TspMig-paloaltonetworks
